virttest/storage.py

```diff
@@ -425,6 +455,36 @@ def auth_info_define_by_params(cls, image, params):
                 image, None, None, storage_type,
                 tls_creds=tls_creds, reconnect_delay=reconnect_delay
             ) if tls_creds or reconnect_delay else None
+        elif enable_curl:
+            # cookie data in a secure way, only for http/https
+            cookie_auth = cls(
+                image, params['curl_cookie_secret'],
+                params.get('curl_cookie_secret_format', 'raw'),
+                storage_type, data_type='cookie'
+            ) if (params['curl_protocol'] in ('http', 'https')
+                  and params.get('curl_cookie_secret')) else None
```

Comment:
I think we don't need to judge whether protocol belongs to http or https since the users have known
the protocol type they want.
```suggestion
            cookie_auth = cls(
                image, params['curl_cookie_secret'],
                params.get('curl_cookie_secret_format', 'raw'),
                storage_type, data_type='cookie'
            ) if params.get('curl_cookie_secret')) else None
```

Reply:
updated